<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<link rel="stylesheet" href="style.css" type="text/css">
<meta content="text/html; charset=iso-8859-1" http-equiv="Content-Type">
<link rel="Start" href="index.html">
<link rel="previous" href="Biocaml_util.html">
<link rel="Up" href="index.html">
<link title="Index of types" rel=Appendix href="index_types.html">
<link title="Index of exceptions" rel=Appendix href="index_exceptions.html">
<link title="Index of values" rel=Appendix href="index_values.html">
<link title="Index of modules" rel=Appendix href="index_modules.html">
<link title="Index of module types" rel=Appendix href="index_module_types.html">
<link title="Biocaml" rel="Chapter" href="Biocaml.html">
<link title="Biocaml_msg" rel="Chapter" href="Biocaml_msg.html">
<link title="Biocaml_pos" rel="Chapter" href="Biocaml_pos.html">
<link title="Biocaml_std" rel="Chapter" href="Biocaml_std.html">
<link title="Biocaml_about" rel="Chapter" href="Biocaml_about.html">
<link title="Biocaml_accu" rel="Chapter" href="Biocaml_accu.html">
<link title="Biocaml_bar" rel="Chapter" href="Biocaml_bar.html">
<link title="Biocaml_bed" rel="Chapter" href="Biocaml_bed.html">
<link title="Biocaml_bpmap" rel="Chapter" href="Biocaml_bpmap.html">
<link title="Biocaml_cel" rel="Chapter" href="Biocaml_cel.html">
<link title="Biocaml_chrName" rel="Chapter" href="Biocaml_chrName.html">
<link title="Biocaml_comments" rel="Chapter" href="Biocaml_comments.html">
<link title="Biocaml_common" rel="Chapter" href="Biocaml_common.html">
<link title="Biocaml_fasta" rel="Chapter" href="Biocaml_fasta.html">
<link title="Biocaml_fastq" rel="Chapter" href="Biocaml_fastq.html">
<link title="Biocaml_genomeMap" rel="Chapter" href="Biocaml_genomeMap.html">
<link title="Biocaml_gff" rel="Chapter" href="Biocaml_gff.html">
<link title="Biocaml_histogram" rel="Chapter" href="Biocaml_histogram.html">
<link title="Biocaml_intervalTree" rel="Chapter" href="Biocaml_intervalTree.html">
<link title="Biocaml_invokeCmdline" rel="Chapter" href="Biocaml_invokeCmdline.html">
<link title="Biocaml_math" rel="Chapter" href="Biocaml_math.html">
<link title="Biocaml_phredScore" rel="Chapter" href="Biocaml_phredScore.html">
<link title="Biocaml_range" rel="Chapter" href="Biocaml_range.html">
<link title="Biocaml_romanNum" rel="Chapter" href="Biocaml_romanNum.html">
<link title="Biocaml_rSet" rel="Chapter" href="Biocaml_rSet.html">
<link title="Biocaml_sam" rel="Chapter" href="Biocaml_sam.html">
<link title="Biocaml_sbml" rel="Chapter" href="Biocaml_sbml.html">
<link title="Biocaml_seq" rel="Chapter" href="Biocaml_seq.html">
<link title="Biocaml_sgr" rel="Chapter" href="Biocaml_sgr.html">
<link title="Biocaml_strandName" rel="Chapter" href="Biocaml_strandName.html">
<link title="Biocaml_table" rel="Chapter" href="Biocaml_table.html">
<link title="Biocaml_track" rel="Chapter" href="Biocaml_track.html">
<link title="Biocaml_tags" rel="Chapter" href="Biocaml_tags.html">
<link title="Biocaml_util" rel="Chapter" href="Biocaml_util.html">
<link title="Biocaml_wig" rel="Chapter" href="Biocaml_wig.html"><title>Biocaml_wig</title>
</head>
<body>
<div class="navbar"><a href="Biocaml_util.html">Previous</a>
&nbsp;<a href="index.html">Up</a>
&nbsp;</div>
<center><h1>Module <a href="type_Biocaml_wig.html">Biocaml_wig</a></h1></center>
<br>
<pre><span class="keyword">module</span> Biocaml_wig: <code class="code">sig</code> <a href="Biocaml_wig.html">..</a> <code class="code">end</code></pre>WIG data.
<p>

    Internal representation of coordinates always assumes the first position on a chromosome is numbered 1. Also, integer ranges are always closed; the range <code class="code">[1, 10]</code> is the set of integers from 1 to 10 inclusive of 1 and 10. WIG data can be in three formats---bed, variable-step, or fixed-step---and unfortunately each has different conventions as follows:<ul>
<li>Bed format requires half-open intervals <code class="code">[low, high\)</code> and numbers the first base as 0. Thus 1 is added to the low value when parsing. The line <code class="code">"chrI 0 10 3.14"</code> is parsed to <code class="code">("chrI", 1, 10, 3.14)</code>.</li>
<li>Variable-step format numbers the first position 1 and uses closed intervals. Thus no change is required. The line <code class="code">"1 3.14"</code> is parsed to <code class="code">(1, 3.14)</code>.</li>
<li>Fixed-step format numbers the first position 1 and uses closed intervals. Thus no change is required. The header line <code class="code">"fixedStep chrom=chrI start=1 step=100 span=30"</code> is parsed to <code class="code">("chrI", 1, 100, 30)</code>.</li>
</ul>

    The inverse is done for printing routines. You are freed from these details if you always use this module to parse and print.
<p>

    All parsers allow columns (fields) on a line to be separated by any combination of space, tab, or carriage return characters. Printers always separate columns with a single tab. Tag-value pairs must be in the form "tag=value" with no space around the '='.<br>
<hr width="100%">
<pre><span id="TYPEpt"><span class="keyword">type</span> <code class="type"></code>pt</span> = <code class="type">string * int * int * float</code> </pre>
<div class="info">
A data point is a 4-tuple <code class="code">(chr,lo,hi,x)</code>, where <code class="code">x</code> is the value assigned to interval <code class="code">[lo, hi]</code>, inclusive of end-points, on chromosome <code class="code">chr</code>.<br>
</div>

<pre><span id="TYPEt"><span class="keyword">type</span> <code class="type"></code>t</span> </pre>
<div class="info">
Type of WIG data. Can be thought of as a collection of <code class="code">pt</code>'s. Coordinates of data points are not allowed to overlap, for each chromosome.<br>
</div>

<br><code><span id="TYPEformat"><span class="keyword">type</span> <code class="type"></code>format</span> = </code><table class="typetable">
<tr>
<td align="left" valign="top" >
<code><span class="keyword">|</span></code></td>
<td align="left" valign="top" >
<code><span class="constructor">Bed</span></code></td>

</tr>
<tr>
<td align="left" valign="top" >
<code><span class="keyword">|</span></code></td>
<td align="left" valign="top" >
<code><span class="constructor">VariableStep</span></code></td>

</tr>
<tr>
<td align="left" valign="top" >
<code><span class="keyword">|</span></code></td>
<td align="left" valign="top" >
<code><span class="constructor">FixedStep</span></code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >The three formats in which WIG data can be specified.</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr></table>


<pre><span id="EXCEPTIONBad"><span class="keyword">exception</span> Bad</span> <span class="keyword">of</span> <code class="type">string</code></pre>
<pre><span id="VALof_list"><span class="keyword">val</span> of_list</span> : <code class="type"><a href="Biocaml_wig.html#TYPEpt">pt</a> list -> <a href="Biocaml_wig.html#TYPEt">t</a></code></pre><div class="info">
Construct WIG data from given <code class="code">pt</code>'s. Raise <code class="code">Bad</code> if any <code class="code">pt</code>'s are invalid.<br>
</div>
<pre><span id="VALto_list"><span class="keyword">val</span> to_list</span> : <code class="type"><a href="Biocaml_wig.html#TYPEt">t</a> -> <a href="Biocaml_wig.html#TYPEpt">pt</a> list</code></pre><div class="info">
Extract data as a flat list.<br>
</div>
<pre><span id="VALiter"><span class="keyword">val</span> iter</span> : <code class="type">(<a href="Biocaml_wig.html#TYPEpt">pt</a> -> unit) -> <a href="Biocaml_wig.html#TYPEt">t</a> -> unit</code></pre><pre><span id="VALfold"><span class="keyword">val</span> fold</span> : <code class="type">('a -> <a href="Biocaml_wig.html#TYPEpt">pt</a> -> 'a) -> 'a -> <a href="Biocaml_wig.html#TYPEt">t</a> -> 'a</code></pre><pre><span id="VALto_file"><span class="keyword">val</span> to_file</span> : <code class="type">?fmt:<a href="Biocaml_wig.html#TYPEformat">format</a> -> <a href="Biocaml_wig.html#TYPEt">t</a> -> string -> unit</code></pre><div class="info">
<code class="code">to_file ~fmt t file</code> prints <code class="code">t</code> to <code class="code">file</code>. Printing is in most efficient format possible by default. Optional <code class="code">fmt</code> argument forces printing in a specific format. Requesting <code class="code">VariableStep</code> or <code class="code">FixedStep</code> may raise <code class="code">Failure</code> if given data cannot be represented in those formats.<br>
</div>
<pre><span id="VALto_channel"><span class="keyword">val</span> to_channel</span> : <code class="type">?fmt:<a href="Biocaml_wig.html#TYPEformat">format</a> -> <a href="Biocaml_wig.html#TYPEt">t</a> -> Pervasives.out_channel -> unit</code></pre><div class="info">
Like <code class="code">to_file</code> but print to channel.<br>
</div>
<pre><span id="VALof_channel"><span class="keyword">val</span> of_channel</span> : <code class="type">?fmt:<a href="Biocaml_wig.html#TYPEformat">format</a> -><br>       ?chr_map:(string -> string) -><br>       ?header:bool -><br>       ?increment_lo_hi:int * int -> Pervasives.in_channel -> <a href="Biocaml_wig.html#TYPEt">t</a></code></pre><pre><span id="VALof_file"><span class="keyword">val</span> of_file</span> : <code class="type">?fmt:<a href="Biocaml_wig.html#TYPEformat">format</a> -><br>       ?chr_map:(string -> string) -><br>       ?header:bool -> ?increment_lo_hi:int * int -> string -> <a href="Biocaml_wig.html#TYPEt">t</a></code></pre></body></html>