<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<link rel="stylesheet" href="style.css" type="text/css">
<meta content="text/html; charset=iso-8859-1" http-equiv="Content-Type">
<link rel="Start" href="index.html">
<link rel="previous" href="Biocaml.Bed.html">
<link rel="next" href="Biocaml.Cel.html">
<link rel="Up" href="Biocaml.html">
<link title="Index of types" rel=Appendix href="index_types.html">
<link title="Index of exceptions" rel=Appendix href="index_exceptions.html">
<link title="Index of values" rel=Appendix href="index_values.html">
<link title="Index of modules" rel=Appendix href="index_modules.html">
<link title="Index of module types" rel=Appendix href="index_module_types.html">
<link title="Biocaml" rel="Chapter" href="Biocaml.html">
<link title="Biocaml_msg" rel="Chapter" href="Biocaml_msg.html">
<link title="Biocaml_pos" rel="Chapter" href="Biocaml_pos.html">
<link title="Biocaml_std" rel="Chapter" href="Biocaml_std.html">
<link title="Biocaml_about" rel="Chapter" href="Biocaml_about.html">
<link title="Biocaml_accu" rel="Chapter" href="Biocaml_accu.html">
<link title="Biocaml_bar" rel="Chapter" href="Biocaml_bar.html">
<link title="Biocaml_bed" rel="Chapter" href="Biocaml_bed.html">
<link title="Biocaml_bpmap" rel="Chapter" href="Biocaml_bpmap.html">
<link title="Biocaml_cel" rel="Chapter" href="Biocaml_cel.html">
<link title="Biocaml_chrName" rel="Chapter" href="Biocaml_chrName.html">
<link title="Biocaml_comments" rel="Chapter" href="Biocaml_comments.html">
<link title="Biocaml_common" rel="Chapter" href="Biocaml_common.html">
<link title="Biocaml_fasta" rel="Chapter" href="Biocaml_fasta.html">
<link title="Biocaml_fastq" rel="Chapter" href="Biocaml_fastq.html">
<link title="Biocaml_genomeMap" rel="Chapter" href="Biocaml_genomeMap.html">
<link title="Biocaml_gff" rel="Chapter" href="Biocaml_gff.html">
<link title="Biocaml_histogram" rel="Chapter" href="Biocaml_histogram.html">
<link title="Biocaml_intervalTree" rel="Chapter" href="Biocaml_intervalTree.html">
<link title="Biocaml_invokeCmdline" rel="Chapter" href="Biocaml_invokeCmdline.html">
<link title="Biocaml_math" rel="Chapter" href="Biocaml_math.html">
<link title="Biocaml_phredScore" rel="Chapter" href="Biocaml_phredScore.html">
<link title="Biocaml_range" rel="Chapter" href="Biocaml_range.html">
<link title="Biocaml_romanNum" rel="Chapter" href="Biocaml_romanNum.html">
<link title="Biocaml_rSet" rel="Chapter" href="Biocaml_rSet.html">
<link title="Biocaml_sam" rel="Chapter" href="Biocaml_sam.html">
<link title="Biocaml_sbml" rel="Chapter" href="Biocaml_sbml.html">
<link title="Biocaml_seq" rel="Chapter" href="Biocaml_seq.html">
<link title="Biocaml_sgr" rel="Chapter" href="Biocaml_sgr.html">
<link title="Biocaml_strandName" rel="Chapter" href="Biocaml_strandName.html">
<link title="Biocaml_table" rel="Chapter" href="Biocaml_table.html">
<link title="Biocaml_track" rel="Chapter" href="Biocaml_track.html">
<link title="Biocaml_tags" rel="Chapter" href="Biocaml_tags.html">
<link title="Biocaml_util" rel="Chapter" href="Biocaml_util.html">
<link title="Biocaml_wig" rel="Chapter" href="Biocaml_wig.html"><link title="I/O" rel="Section" href="#6_IO">
<title>Biocaml.Bpmap</title>
</head>
<body>
<div class="navbar"><a href="Biocaml.Bed.html">Previous</a>
&nbsp;<a href="Biocaml.html">Up</a>
&nbsp;<a href="Biocaml.Cel.html">Next</a>
</div>
<center><h1>Module <a href="type_Biocaml.Bpmap.html">Biocaml.Bpmap</a></h1></center>
<br>
<pre><span class="keyword">module</span> Bpmap: <code class="type">Biocaml_bpmap</code></pre><hr width="100%">
<br><code><span id="TYPEprobe"><span class="keyword">type</span> <code class="type"></code>probe</span> = {</code><table class="typetable">
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>org_name&nbsp;: <code class="type">string</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >name of organism on which probe is based</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>version&nbsp;: <code class="type">string</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >genome build version on which probe is based</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>chr_name&nbsp;: <code class="type">string</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >name of chromosome on which probe is based</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>start_pos&nbsp;: <code class="type">int</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >start position of probe on given chromosome</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>sequence&nbsp;: <code class="type"><a href="Biocaml_seq.html#TYPEt">Biocaml_seq.t</a></code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >sequence of the perfect match probe</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr></table>
}


<br><code><span id="TYPErow"><span class="keyword">type</span> <code class="type"></code>row</span> = {</code><table class="typetable">
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>pmcoord&nbsp;: <code class="type">int * int</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >x,y-coordinates of perfect match probe.</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>mmcoord&nbsp;: <code class="type">int * int</code>;</code></td>
<td class="typefieldcomment" align="left" valign="top" ><code>(*</code></td><td class="typefieldcomment" align="left" valign="top" >x,y-coordinates of mis-match probe</td><td class="typefieldcomment" align="left" valign="bottom" ><code>*)</code></td>
</tr>
<tr>
<td align="left" valign="top" >
<code>&nbsp;&nbsp;</code></td>
<td align="left" valign="top" >
<code>probe&nbsp;: <code class="type"><a href="Biocaml_bpmap.html#TYPEprobe">probe</a></code>;</code></td>

</tr></table>
}

<div class="info">
Type of information on one data row.<br>
</div>

<pre><span id="TYPEt"><span class="keyword">type</span> <code class="type"></code>t</span> </pre>
<div class="info">
Type of a BPMAP file.<br>
</div>

<pre><span id="EXCEPTIONBad"><span class="keyword">exception</span> Bad</span> <span class="keyword">of</span> <code class="type">string</code></pre>
<div class="info">
Raised when encountering ill-formed BPMAP.<br>
</div>
<pre><span id="VALnum_probes"><span class="keyword">val</span> num_probes</span> : <code class="type"><a href="Biocaml_bpmap.html#TYPEt">t</a> -> int</code></pre><div class="info">
Number of PM/MM probe pairs in given BPMAP. The number of total probes is twice this value.<br>
</div>
<pre><span id="VALcol_names"><span class="keyword">val</span> col_names</span> : <code class="type">string list</code></pre><div class="info">
Names of columns in BPMAP file, in the order required by specification.<br>
</div>
<pre><span id="VALiter"><span class="keyword">val</span> iter</span> : <code class="type">(<a href="Biocaml_bpmap.html#TYPErow">row</a> -> unit) -> <a href="Biocaml_bpmap.html#TYPEt">t</a> -> unit</code></pre><pre><span id="VALfold"><span class="keyword">val</span> fold</span> : <code class="type">('a -> <a href="Biocaml_bpmap.html#TYPErow">row</a> -> 'a) -> 'a -> <a href="Biocaml_bpmap.html#TYPEt">t</a> -> 'a</code></pre><pre><span id="VALto_list"><span class="keyword">val</span> to_list</span> : <code class="type"><a href="Biocaml_bpmap.html#TYPEt">t</a> -> <a href="Biocaml_bpmap.html#TYPErow">row</a> list</code></pre><br>
<span id="6_IO"><h6>I/O</h6></span><br>
<pre><span id="VALof_file"><span class="keyword">val</span> of_file</span> : <code class="type">?chr_map:(string -> string) -> string -> <a href="Biocaml_bpmap.html#TYPEt">t</a></code></pre><div class="info">
<code class="code">of_file file</code> parses <code class="code">file</code>. If given, <code class="code">chr_map</code> is applied to every chromosome name. Raise <code class="code">Bad</code> if there is a parse error.<br>
</div>
<pre><span id="VALrow_to_string"><span class="keyword">val</span> row_to_string</span> : <code class="type"><a href="Biocaml_bpmap.html#TYPErow">row</a> -> string</code></pre><div class="info">
String representation of row in same format as required by specification.<br>
</div>
<pre><span id="VALto_file"><span class="keyword">val</span> to_file</span> : <code class="type">string -> <a href="Biocaml_bpmap.html#TYPEt">t</a> -> unit</code></pre><div class="info">
<code class="code">to_file file t</code> prints <code class="code">t</code> to <code class="code">file</code> in format required by specification.<br>
</div>
</body></html>