<html>
<head>
<link rel="stylesheet" href="style.css" type="text/css">
<meta content="text/html; charset=iso-8859-1" http-equiv="Content-Type">
<link rel="Start" href="index.html">
<link title="Index of types" rel=Appendix href="index_types.html">
<link title="Index of exceptions" rel=Appendix href="index_exceptions.html">
<link title="Index of values" rel=Appendix href="index_values.html">
<link title="Index of modules" rel=Appendix href="index_modules.html">
<link title="Index of module types" rel=Appendix href="index_module_types.html">
<link title="Biocaml" rel="Chapter" href="Biocaml.html">
<link title="Biocaml_msg" rel="Chapter" href="Biocaml_msg.html">
<link title="Biocaml_pos" rel="Chapter" href="Biocaml_pos.html">
<link title="Biocaml_std" rel="Chapter" href="Biocaml_std.html">
<link title="Biocaml_about" rel="Chapter" href="Biocaml_about.html">
<link title="Biocaml_accu" rel="Chapter" href="Biocaml_accu.html">
<link title="Biocaml_bar" rel="Chapter" href="Biocaml_bar.html">
<link title="Biocaml_bed" rel="Chapter" href="Biocaml_bed.html">
<link title="Biocaml_bpmap" rel="Chapter" href="Biocaml_bpmap.html">
<link title="Biocaml_cel" rel="Chapter" href="Biocaml_cel.html">
<link title="Biocaml_chrName" rel="Chapter" href="Biocaml_chrName.html">
<link title="Biocaml_comments" rel="Chapter" href="Biocaml_comments.html">
<link title="Biocaml_common" rel="Chapter" href="Biocaml_common.html">
<link title="Biocaml_fasta" rel="Chapter" href="Biocaml_fasta.html">
<link title="Biocaml_fastq" rel="Chapter" href="Biocaml_fastq.html">
<link title="Biocaml_genomeMap" rel="Chapter" href="Biocaml_genomeMap.html">
<link title="Biocaml_gff" rel="Chapter" href="Biocaml_gff.html">
<link title="Biocaml_histogram" rel="Chapter" href="Biocaml_histogram.html">
<link title="Biocaml_intervalTree" rel="Chapter" href="Biocaml_intervalTree.html">
<link title="Biocaml_invokeCmdline" rel="Chapter" href="Biocaml_invokeCmdline.html">
<link title="Biocaml_math" rel="Chapter" href="Biocaml_math.html">
<link title="Biocaml_phredScore" rel="Chapter" href="Biocaml_phredScore.html">
<link title="Biocaml_range" rel="Chapter" href="Biocaml_range.html">
<link title="Biocaml_romanNum" rel="Chapter" href="Biocaml_romanNum.html">
<link title="Biocaml_rSet" rel="Chapter" href="Biocaml_rSet.html">
<link title="Biocaml_sam" rel="Chapter" href="Biocaml_sam.html">
<link title="Biocaml_sbml" rel="Chapter" href="Biocaml_sbml.html">
<link title="Biocaml_seq" rel="Chapter" href="Biocaml_seq.html">
<link title="Biocaml_sgr" rel="Chapter" href="Biocaml_sgr.html">
<link title="Biocaml_strandName" rel="Chapter" href="Biocaml_strandName.html">
<link title="Biocaml_table" rel="Chapter" href="Biocaml_table.html">
<link title="Biocaml_track" rel="Chapter" href="Biocaml_track.html">
<link title="Biocaml_tags" rel="Chapter" href="Biocaml_tags.html">
<link title="Biocaml_util" rel="Chapter" href="Biocaml_util.html">
<link title="Biocaml_wig" rel="Chapter" href="Biocaml_wig.html"><title>Index of values</title>
</head>
<body>
<center><h1>Index of values</h1></center>
<table>
<tr><td align="left"><br></td></tr>
<tr><td><a href="Biocaml_std.html#VAL(-|)">(-|)</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.html#VAL(<<-)">(&lt;&lt;-)</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Option.With_monad.html#VAL(>>=)">(&gt;&gt;=)</a> [<a href="Biocaml_std.Option.With_monad.html">Biocaml_std.Option.With_monad</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.html#VAL(|-)">(|-)</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.html#VAL(|>)">(|&gt;)</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td align="left"><br>A</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALadd">add</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td><div class="info">
<code class="code">add lo hi v t</code> adds the interval (<code class="code">lo</code>, <code class="code">hi</code>) labeled with value <code class="code">v</code> to the contents of <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALadd">add</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALadd">add</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_accu.Counter.html#VALadd">add</a> [<a href="Biocaml_accu.Counter.html">Biocaml_accu.Counter</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_accu.html#VALadd">add</a> [<a href="Biocaml_accu.html">Biocaml_accu</a>]</td>
<td><div class="info">
<code class="code">add accu x y</code> updates the value in <code class="code">accu</code> for
    the bin of <code class="code">x</code> by an increment <code class="code">y</code>
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALadd_attribute">add_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">add_attribute x y r</code> adds attribute <code class="code">x</code> with value <code class="code">y</code> in
      <code class="code">r</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.Tree.html#VALadd_child">add_child</a> [<a href="Biocaml_msg.Tree.html">Biocaml_msg.Tree</a>]</td>
<td><div class="info">
<code class="code">add_child t x</code> inserts <code class="code">x</code> as the right-most child of <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.StringMap.html#VALadd_with">add_with</a> [<a href="Biocaml_std.StringMap.html">Biocaml_std.StringMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALafter">after</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">after u v</code> is equivalent to <code class="code">before v u</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALalg_name">alg_name</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Name of algorithm used to generate scores.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALalg_version">alg_version</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Version number of algorithm used.
</div>
</td></tr>
<tr><td><a href="Biocaml_sam.html#VALalignment_to_string">alignment_to_string</a> [<a href="Biocaml_sam.html">Biocaml_sam</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALall_positional">all_positional</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Return true if all pairs of given ranges are positionally comparable.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALany_overlap">any_overlap</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Return true if any pair of given ranges overlap each other.
</div>
</td></tr>
<tr><td><a href="Biocaml_chrName.html#VALarabic">arabic</a> [<a href="Biocaml_chrName.html">Biocaml_chrName</a>]</td>
<td><div class="info">
<code class="code">arabic s</code> returns the canonical Arabic representation of <code class="code">s</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.List.html#VALassoc_exn">assoc_exn</a> [<a href="Biocaml_std.List.html">Biocaml_std.List</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALattribute_names">attribute_names</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
List of attributes defined for the given <code class="code">row</code>.
</div>
</td></tr>
<tr><td align="left"><br>B</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALbackwards">backwards</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALbefore">before</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">before u v</code> is true if <code class="code">strict_before u v || equal u v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALbin">bin</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">bin hist i</code> returns the <code class="code">i</code>th bin of <code class="code">hist</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.html#VALbug">bug</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td><div class="info">
Create a string communicating an error, warning, or bug.
</div>
</td></tr>
<tr><td align="left"><br>C</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALcardinal">cardinal</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALcheck_integrity">check_integrity</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td><div class="info">
Used for debugging purpose, should be removed in the long run
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALcnd">cnd</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Cumulative distribution function.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALcol_exn">col_exn</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
Return the file name, line number, or column number.
</div>
</td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALcol_names">col_names</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td><div class="info">
Names of columns in BPMAP file, in the order required by specification.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALcolumn">column</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">column m i</code> extracts the <code class="code">i</code>th column of matrix <code class="code">m</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_comments.html#VALcomment_char">comment_char</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
Return the start character for the given comments.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALcompare">compare</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALcompare_containment">compare_containment</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">compare_containment u v</code> returns -1 if <code class="code">u</code> is a strict subset of <code class="code">v</code>, 0 if <code class="code">u</code> is equal to <code class="code">v</code>, +1 if <code class="code">u</code> is a strict superset of <code class="code">v</code>, and returns None otherwise.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALcompare_hi">compare_hi</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Compare by ranges' <code class="code">hi</code> bounds, ignoring <code class="code">lo</code> bounds.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALcompare_lo">compare_lo</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Compare by ranges' <code class="code">lo</code> bounds, ignoring <code class="code">hi</code> bounds.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALcompare_lo_then_hi">compare_lo_then_hi</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Return order between <code class="code">lo</code> bounds unless they are equal, in which case return order between <code class="code">hi</code> bounds.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALcompare_positional">compare_positional</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">compare_positional u v</code> returns -1 if <code class="code">u</code> is strictly before <code class="code">v</code>, 0 if <code class="code">u</code> is equal to <code class="code">v</code>, +1 if <code class="code">u</code> is strictly after <code class="code">v</code>, and returns None otherwise.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Order.html#VALcompose">compose</a> [<a href="Biocaml_std.Order.html">Biocaml_std.Order</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALconcat">concat</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td><div class="info">
<code class="code">concat t1 t2</code> returns a type representing the lines <code class="code">t1</code> followed by <code class="code">t2</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_comments.html#VALconcat">concat</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
<code class="code">concat a b</code> concatenates comment lines <code class="code">a</code> and <code class="code">b</code> such that
      <code class="code">a</code> is followed by <code class="code">b</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALcoord_convention">coord_convention</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Probe coordinate convention used.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALcopy">copy</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Copy histogram.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALcount">count</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">count hist i</code> returns the count the <code class="code">i</code>th bin.
</div>
</td></tr>
<tr><td><a href="Biocaml_accu.html#VALcounts">counts</a> [<a href="Biocaml_accu.html">Biocaml_accu</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_accu.Counter.html#VALcreate">create</a> [<a href="Biocaml_accu.Counter.html">Biocaml_accu.Counter</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_accu.html#VALcreate">create</a> [<a href="Biocaml_accu.html">Biocaml_accu</a>]</td>
<td><div class="info">
<code class="code">create ~n e f op</code> creates an accumulator, which maps instances to bins
    with <code class="code">f</code>, uses <code class="code">e</code> as a neutral element (that is the value associated to a 
    bin before any value has been added to it) and updates the value of a bin 
    with <code class="code">op</code>.
</div>
</td></tr>
<tr><td align="left"><br>D</td></tr>
<tr><td><a href="Biocaml_cel.html#VALdata">data</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
<code class="code">data bpmap cels</code> returns a list associating probes with pairs of (PM,MM) idata in each of the given cel files (in the same order of course).
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALdata_type">data_type</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Return the type of data, either "signal" or "p-value".
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.Test.html#VALdefault_test">default_test</a> [<a href="Biocaml_rSet.Test.html">Biocaml_rSet.Test</a>]</td>
<td><div class="info">
This function generates random lists and uses them as arguments for <code class="code">test</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALdelete_attribute">delete_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">delete_attribute x r</code> deletes all occurrences of attribute <code class="code">x</code>
      in <code class="code">r</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALdense">dense</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_rSet.html#VALdiff">diff</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Set difference.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALdiff">diff</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td align="left"><br>E</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALelements">elements</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALempty">empty</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td><div class="info">
the empty tree
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALempty">empty</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
Empty track line.
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.html#VALempty">empty</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
The empty set.
</div>
</td></tr>
<tr><td><a href="Biocaml_comments.html#VALempty">empty</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
<code class="code">empty c</code> is the empty list of comment lines, but with the
      specification that <code class="code">c</code> will be used as the start character.
</div>
</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALenum">enum</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALenum">enum</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALenum">enum</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALenum">enum</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.Signal.html#VALenum">enum</a> [<a href="Biocaml_genomeMap.Signal.html">Biocaml_genomeMap.Signal</a>]</td>
<td><div class="info">
enumeration over all constant intervals of the function, in increasing order
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALenum">enum</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_comments.html#VALenum">enum</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
Return start character and enumeration of comment lines in given
      <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_accu.html#VALenum">enum</a> [<a href="Biocaml_accu.html">Biocaml_accu</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sam.html#VALenum_file">enum_file</a> [<a href="Biocaml_sam.html">Biocaml_sam</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_fastq.html#VALenum_input">enum_input</a> [<a href="Biocaml_fastq.html">Biocaml_fastq</a>]</td>
<td><div class="info">
Returns enumeration of fastq records in given input.
</div>
</td></tr>
<tr><td><a href="Biocaml_fasta.html#VALenum_input">enum_input</a> [<a href="Biocaml_fasta.html">Biocaml_fasta</a>]</td>
<td><div class="info">
Returns comments and enumeration of fasta records in given input.
</div>
</td></tr>
<tr><td><a href="Biocaml_bed.html#VALenum_input">enum_input</a> [<a href="Biocaml_bed.html">Biocaml_bed</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALepsilon">epsilon</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">epsilon f init fin</code> applies <code class="code">f n fin</code> to all numbers from <code class="code">init</code> to <code class="code">fin</code> and adds them up.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALequal">equal</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_msg.html#VALerr">err</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_util.html#VALerr_enum">err_enum</a> [<a href="Biocaml_util.html">Biocaml_util</a>]</td>
<td><div class="info">
Returned enum behaves identically to given one except that its
      use will raise <code class="code">PosError</code> when accessing its element raises an
      exception <code class="code">e</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Signal.html#VALeval">eval</a> [<a href="Biocaml_genomeMap.Signal.html">Biocaml_genomeMap.Signal</a>]</td>
<td><div class="info">
function evaluation at some point in the genome
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALeven">even</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
True if given integer is even.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.String.html#VALexists">exists</a> [<a href="Biocaml_std.String.html">Biocaml_std.String</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALexpand_assoc_list">expand_assoc_list</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">exp_assoc_list dat</code> returns a list associating each integer <code class="code">i</code> with the list of values associated with all ranges overlapping <code class="code">i</code> in <code class="code">dat</code>.
</div>
</td></tr>
<tr><td align="left"><br>F</td></tr>
<tr><td><a href="Biocaml_pos.html#VALf">f</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALfactorial">factorial</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Self-explanatory.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALfile_exn">file_exn</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_tags.html#VALfind">find</a> [<a href="Biocaml_tags.html">Biocaml_tags</a>]</td>
<td><div class="info">
<code class="code">find x t</code> returns the value of tag <code class="code">x</code> in <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALfind">find</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
<code class="code">find t a</code> returns the value of attribute <code class="code">a</code>, or raises <code class="code">Not_found</code> if no such attribute.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALfind_bin_index">find_bin_index</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">find_bin_index hist x</code> returns the index of the bin in <code class="code">hist</code> containing <code class="code">x</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALfind_closest">find_closest</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td><div class="info">
<code class="code">find_closest lo hi t</code> returns the interval in <code class="code">t</code> which is at minimal
    distance of the interval (<code class="code">lo</code>,<code class="code">hi</code>).
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALfind_min_range">find_min_range</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">find_min_range v pred i</code> finds the minimum sized range within <code class="code">v</code> centered around <code class="code">i</code> that satisfies <code class="code">pred</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALfind_min_window">find_min_window</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">find_min_window a pred i</code> finds the minimum sized window within <code class="code">a</code> centered around index <code class="code">i</code> that satisfies <code class="code">pred</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALfind_regions">find_regions</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
TO DO: fill in this documentation.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALfind_regions">find_regions</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">find_regions ~max_gap pred a</code> returns an array of <code class="code">(first,last)</code> index pairs denoting boundaries (inclusive) of regions found in <code class="code">a</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALfl">fl</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_pos.html#VALflc">flc</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
Methods for creating a position.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.html#VALflip">flip</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALfold">fold</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALfold">fold</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALfold">fold</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALfold">fold</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td><div class="info">
fold guaranteed on increasing order keywise, and for each key
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALfold">fold</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td><div class="info">
fold guaranteed on increasing order keywise, and for each key
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Signal.html#VALfold">fold</a> [<a href="Biocaml_genomeMap.Signal.html">Biocaml_genomeMap.Signal</a>]</td>
<td><div class="info">
folds on constant intervals of the function, in increasing order
</div>
</td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALfold">fold</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALfold_channel">fold_channel</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALfold_channel'">fold_channel'</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALfold_file">fold_file</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALfold_file">fold_file</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">fold_file f init file</code> accumulates the result of applying <code class="code">f</code>
      to each row of <code class="code">file</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALfold_left">fold_left</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
see String.fold_left.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.String.html#VALfold_lefti">fold_lefti</a> [<a href="Biocaml_std.String.html">Biocaml_std.String</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_seq.html#VALfold_lefti">fold_lefti</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
String.fold_lefti
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALfold_stream">fold_stream</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALfold_stream'">fold_stream'</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.String.html#VALfor_all">for_all</a> [<a href="Biocaml_std.String.html">Biocaml_std.String</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALfull">full</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td><div class="info">
<code class="code">hide</code>, <code class="code">dense</code>, <code class="code">pack</code>, <code class="code">squish</code>, and <code class="code">full</code> each take a string that should either be "all" or a space separated list of track names.
</div>
</td></tr>
<tr><td align="left"><br>G</td></tr>
<tr><td><a href="Biocaml_range.html#VALgap">gap</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">gap u v</code> returns the size of the gap between <code class="code">u</code> and <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALgenomic_map">genomic_map</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
File path of bpmap file used to generate scores.
</div>
</td></tr>
<tr><td><a href="Biocaml_accu.html#VALget">get</a> [<a href="Biocaml_accu.html">Biocaml_accu</a>]</td>
<td><div class="info">
<code class="code">get accu x</code> returns the value associated to <code class="code">b</code> in <code class="code">accu</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALget_attribute">get_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">get_attribute r x</code> returns the value of attribute
      <code class="code">x</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALget_attributel">get_attributel</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">get_attributel r x</code> returns the values of the attribute named
      <code class="code">x</code> in row <code class="code">r</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Test.html#VALget_time">get_time</a> [<a href="Biocaml_std.Test.html">Biocaml_std.Test</a>]</td>
<td></td></tr>
<tr><td align="left"><br>H</td></tr>
<tr><td><a href="Biocaml_gff.html#VALhas_attribute">has_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">has_attribute r x</code> returns true if attribute <code class="code">x</code> is defined in <code class="code">r</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.List.html#VALhd_exn">hd_exn</a> [<a href="Biocaml_std.List.html">Biocaml_std.List</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sam.html#VALheader_line_to_string">header_line_to_string</a> [<a href="Biocaml_sam.html">Biocaml_sam</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sam.html#VALheader_to_string">header_to_string</a> [<a href="Biocaml_sam.html">Biocaml_sam</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALhide">hide</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALhistogram">histogram</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Return histogram of values using <code class="code">cmp</code> (default = <code class="code">Pervasives.compare</code>) for comparison.
</div>
</td></tr>
<tr><td align="left"><br>I</td></tr>
<tr><td><a href="Biocaml_std.html#VALidentity">identity</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALidxsort">idxsort</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">idxsort cmp a</code> is like <code class="code">Array.sort</code> but <code class="code">a</code> is unaltered, and instead an array of the indices in sorted order is returned.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALifold">ifold</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
<code class="code">ifold f a t</code> folds over data rows in intensity section of <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALiiter">iiter</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
<code class="code">iiter f t</code> iterates over the rows in intensity section of <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALin_range">in_range</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">in_range hist x</code> is true if <code class="code">x</code> greater than or equal to <code class="code">minimum hist</code> and strictly less than <code class="code">maximum hist</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_sbml.html#VALin_sbml">in_sbml</a> [<a href="Biocaml_sbml.html">Biocaml_sbml</a>]</td>
<td><div class="info">
Returns an sb_model read from input stream
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALincrement">increment</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">increment delt hist x</code> increments the count of the bin containing <code class="code">x</code> by <code class="code">delt</code> (default is 1.0).
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALincrl">incrl</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
<code class="code">incrl pos k</code> increments the line number of <code class="code">pos</code> by <code class="code">k</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALindex_by_attribute">index_by_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">index_by_attribute x</code> returns a mapping from the values of
      attribute <code class="code">x</code> to those rows with that value.
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.html#VALinter">inter</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Set intersection.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALinter">inter</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALintersect">intersect</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">intersect u v</code> returns the range representing the intersection of <code class="code">u</code> and <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALintersection_size">intersection_size</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALintersects">intersects</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALintersects">intersects</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALintersects">intersects</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_invokeCmdline.html#VALinvoke_cmd">invoke_cmd</a> [<a href="Biocaml_invokeCmdline.html">Biocaml_invokeCmdline</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Char.html#VALis_alpha_num">is_alpha_num</a> [<a href="Biocaml_std.Char.html">Biocaml_std.Char</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_comments.html#VALis_comments">is_comments</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
Like <code class="code">of_string</code> but simply returns true if given string can be
      parsed as valid comment lines.
</div>
</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALis_empty">is_empty</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALis_empty">is_empty</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_rSet.html#VALis_empty">is_empty</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Return true if given set is empty.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALis_nucleic_acid">is_nucleic_acid</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
True if given character represents one of the allowed nucleic acid codes, case-insensitive.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Array.html#VALis_rectangular">is_rectangular</a> [<a href="Biocaml_std.Array.html">Biocaml_std.Array</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Char.html#VALis_space">is_space</a> [<a href="Biocaml_std.Char.html">Biocaml_std.Char</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALiter">iter</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALiter">iter</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALiter">iter</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALiter_file">iter_file</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALiter_file">iter_file</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td align="left"><br>K</td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALkeep_while">keep_while</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALkeep_whilei">keep_whilei</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td align="left"><br>L</td></tr>
<tr><td><a href="Biocaml_pos.html#VALl">l</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_pos.html#VALlc">lc</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_msg.Tree.html#VALleaf">leaf</a> [<a href="Biocaml_msg.Tree.html">Biocaml_msg.Tree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_seq.html#VALlength">length</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
Length of sequence.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALline_exn">line_exn</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALlines_of_channel">lines_of_channel</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALlines_of_chars">lines_of_chars</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALlog">log</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Logarithm.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALlog10">log10</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Base 10 logarithm.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALlog2">log2</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Base 2 logarithm.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALltqnorm">ltqnorm</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Lower tail quantile for standard normal distribution function.
</div>
</td></tr>
<tr><td align="left"><br>M</td></tr>
<tr><td><a href="Biocaml_math.html#VALmad">mad</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Median absolute deviation (MAD).
</div>
</td></tr>
<tr><td><a href="Biocaml_about.html#VALmajor_version">major_version</a> [<a href="Biocaml_about.html">Biocaml_about</a>]</td>
<td><div class="info">
Major version number.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Tuple.Tr.html#VALmake">make</a> [<a href="Biocaml_std.Tuple.Tr.html">Biocaml_std.Tuple.Tr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Tuple.Pr.html#VALmake">make</a> [<a href="Biocaml_std.Tuple.Pr.html">Biocaml_std.Tuple.Pr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALmake">make</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">make l u</code> returns the range from <code class="code">l</code> to <code class="code">u</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALmake">make</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">make cmp bins</code> returns a new histogram from the given <code class="code">bins</code>, all initialized to a count of 0.0.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALmake">make</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALmake">make</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.Signal.html#VALmake">make</a> [<a href="Biocaml_genomeMap.Signal.html">Biocaml_genomeMap.Signal</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALmake_opt">make_opt</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Like <code class="code">make</code> but returns None instead of raising exception.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALmake_uniform">make_uniform</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
<code class="code">make_uniform min max n</code> returns a histogram with <code class="code">n</code> bins uniformly dividing up the range from <code class="code">min</code> to <code class="code">max</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALmap">map</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALmap_of_file">map_of_file</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sbml.html#VALmath_to_string">math_to_string</a> [<a href="Biocaml_sbml.html">Biocaml_sbml</a>]</td>
<td><div class="info">
Returns a string with sb_math converted into a S-expression
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALmax">max</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Return maximum value in given array.
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.html#VALmax_array_length_error">max_array_length_error</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td><div class="info">
String explaining OCaml's array length limitation on 32-bit machines.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALmax_gap_of_positional">max_gap_of_positional</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
Return maximum gap between adjacent pairs of given ranges.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALmaximum">maximum</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Upper limit of the maximum bin.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALmean">mean</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Mean.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALmedian">median</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Median.
</div>
</td></tr>
<tr><td><a href="Biocaml_tags.html#VALmem">mem</a> [<a href="Biocaml_tags.html">Biocaml_tags</a>]</td>
<td><div class="info">
<code class="code">mem x t</code> returns true if <code class="code">t</code> contains a value for tag <code class="code">x</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALmem">mem</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
<code class="code">mem t a x</code> returns true if <code class="code">a</code> is assigned value <code class="code">x</code> in <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALmember">member</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">member t k</code> returns true if <code class="code">t</code> contains <code class="code">k</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALmin">min</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Return minimum value in given array.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALminimum">minimum</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Lower limit of the minimum bin.
</div>
</td></tr>
<tr><td><a href="Biocaml_about.html#VALminor_version">minor_version</a> [<a href="Biocaml_about.html">Biocaml_about</a>]</td>
<td><div class="info">
Minor version number.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALminus_plus">minus_plus</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td><div class="info">
Return "-" or "+".
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALmm">mm</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
Similar to <a href="Biocaml_cel.html#VALdata"><code class="code">Biocaml_cel.data</code></a> but the data returned are the MM mean intensity values in <code class="code">cels</code>.
</div>
</td></tr>
<tr><td align="left"><br>N</td></tr>
<tr><td><a href="Biocaml_std.List.html#VALnpartition_exn">npartition_exn</a> [<a href="Biocaml_std.List.html">Biocaml_std.List</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_seq.html#VALnth">nth</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
<code class="code">nth t i</code> returns the <code class="code">i</code>th nucleic acid in sequence <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALnum_bins">num_bins</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Number of bins.
</div>
</td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALnum_probes">num_probes</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td><div class="info">
Number of PM/MM probe pairs in given BPMAP.
</div>
</td></tr>
<tr><td align="left"><br>O</td></tr>
<tr><td><a href="Biocaml_math.html#VALodd">odd</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
True if given integer is odd.
</div>
</td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALof_ascii">of_ascii</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td><div class="info">
<code class="code">of_ascii ~offset x</code> returns the PHRED score encoded by ASCII
      character <code class="code">x</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALof_buffer">of_buffer</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
Make sequence from buffer.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALof_buffer_unsafe">of_buffer_unsafe</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALof_channel">of_channel</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALof_channel">of_channel</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALof_channel">of_channel</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALof_chr_lists">of_chr_lists</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALof_file">of_file</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALof_file">of_file</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALof_file">of_file</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">of_file file</code> parses <code class="code">file</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALof_file">of_file</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
Parse given file if possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALof_file">of_file</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td><div class="info">
<code class="code">of_file file</code> parses <code class="code">file</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALof_file">of_file</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Parse file.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALof_file_opt">of_file_opt</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
Parse given file if possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_table.html#VALof_input">of_input</a> [<a href="Biocaml_table.html">Biocaml_table</a>]</td>
<td><div class="info">
Default itags is:
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALof_int">of_int</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
Convert integer to roman numeral if possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALof_int">of_int</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td><div class="info">
<code class="code">of_int x</code> returns the PHRED score with the same value <code class="code">x</code>, but
      assures that <code class="code">x</code> is non-negative.
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALof_int_exn">of_int_exn</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
Like <code class="code">of_int</code> but raise <code class="code">Bad</code> if unable to convert integer.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.IntSet.html#VALof_list">of_list</a> [<a href="Biocaml_std.IntSet.html">Biocaml_std.IntSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALof_list">of_list</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td><div class="info">
Construct WIG data from given <code class="code">pt</code>'s.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALof_list">of_list</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
<code class="code">of_list l</code> sets <code class="code">a</code> to <code class="code">x</code> for every <code class="code">(a,x)</code> pair in <code class="code">l</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.html#VALof_list">of_list</a> [<a href="Biocaml_track.html">Biocaml_track</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALof_list">of_list</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALof_list">of_list</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALof_locations">of_locations</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td><div class="info">
<code class="code">of_locations e</code> computes a selection as the union of the locations contained in <code class="code">e</code>
</div>
</td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALof_probability">of_probability</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td><div class="info">
<code class="code">of_probability ~f x</code> returns <code class="code">-10 * log_10(x)</code>, which is the
      definition of PHRED scores.
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.html#VALof_range_list">of_range_list</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Construct the set of integers representing the union of integers in all given ranges.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALof_stream">of_stream</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_tags.html#VALof_string">of_string</a> [<a href="Biocaml_tags.html">Biocaml_tags</a>]</td>
<td><div class="info">
Parse given string as list of tags.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALof_string">of_string</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td><div class="info">
Creates a list of lines by splitting given string on newline characters.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALof_string">of_string</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
Parse string to track line.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALof_string">of_string</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td><div class="info">
Convert string to chromosome name if possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALof_string">of_string</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
Make sequence from string.
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALof_string">of_string</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
Parse string as a roman numeral if possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_comments.html#VALof_string">of_string</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
<code class="code">of_string c s</code> creates list of lines by splitting given string
      on newline characters.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALof_string_exn">of_string_exn</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td><div class="info">
Like <code class="code">of_string</code> but raise <code class="code">Bad</code> if conversion not possible.
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALof_string_exn">of_string_exn</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
Like <code class="code">of_string</code> but raise <code class="code">Bad</code> if unable to parse string.
</div>
</td></tr>
<tr><td><a href="Biocaml_table.html#VALof_string_list">of_string_list</a> [<a href="Biocaml_table.html">Biocaml_table</a>]</td>
<td><div class="info">
Default <code class="code">itags</code> are:
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALof_string_unsafe">of_string_unsafe</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
Make a sequence from given buffer or string.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.html#VALopen_out_safe">open_out_safe</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.html#VALoutput_endline">output_endline</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_range.html#VALoverlap">overlap</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">overlap u v</code> returns amount of overlap between two ranges.
</div>
</td></tr>
<tr><td align="left"><br>P</td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALpack">pack</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_invokeCmdline.html#VALparse_lines">parse_lines</a> [<a href="Biocaml_invokeCmdline.html">Biocaml_invokeCmdline</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALpearson">pearson</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">pearson arr1 arr2</code> computes the Pearson product-moment correlation coefficient of two float arrays.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALpm">pm</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
Similar to <a href="Biocaml_cel.html#VALdata"><code class="code">Biocaml_cel.data</code></a> but the data returned are the PM mean intensity values in <code class="code">cels</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_cel.html#VALpm_mm">pm_mm</a> [<a href="Biocaml_cel.html">Biocaml_cel</a>]</td>
<td><div class="info">
Similar to <a href="Biocaml_cel.html#VALdata"><code class="code">Biocaml_cel.data</code></a> but the data returned are the PM-MM mean intensity values in <code class="code">cels</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALposition">position</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td><div class="info">
<code class="code">position chr start end</code> returns the browser line "chr:start-end".
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALprediction_values">prediction_values</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">prediction_values tp tn fp fn</code> takes 4 arguments: the number of true-positives <code class="code">tp</code>, true-negatives <code class="code">tn</code>, false-positives <code class="code">fp</code>, and false-negatives <code class="code">fn</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_intervalTree.html#VALprint">print</a> [<a href="Biocaml_intervalTree.html">Biocaml_intervalTree</a>]</td>
<td><div class="info">
Used for debugging purpose, should be removed in the long run
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.html#VALprint_bug">print_bug</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td><div class="info">
Print an error, warning, or bug.
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.html#VALprint_err">print_err</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_msg.html#VALprint_warn">print_warn</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALpseudomedian">pseudomedian</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Pseudomedian is the median of all pairwise averages of values in given array (not including self-pairs).
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALpwfold">pwfold</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td><div class="info">
a constant interval of an LMap <code class="code">lm</code> is a range intersecting <code class="code">lm</code> and
that do not contain any start or end of a range from <code class="code">lm</code>.
</div>
</td></tr>
<tr><td align="left"><br>Q</td></tr>
<tr><td><a href="Biocaml_math.html#VALquantile_normalization">quantile_normalization</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Input matrix <code class="code">m</code> should be arranged such that <code class="code">m.(i).(j)</code> is the <code class="code">i</code>th measurement in experiment <code class="code">j</code>.
</div>
</td></tr>
<tr><td align="left"><br>R</td></tr>
<tr><td><a href="Biocaml_std.Lines.html#VALraise_error">raise_error</a> [<a href="Biocaml_std.Lines.html">Biocaml_std.Lines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALrange">range</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">range step first last</code> returns array [|first; first +.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALrange_floats">range_floats</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALrange_ints">range_ints</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALrank">rank</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">rank arr</code> returns an array of ranked values, where ties are given the mean of what the rank would otherwise be.
</div>
</td></tr>
<tr><td><a href="Biocaml_sam.html#VALrecord_type_code_of_header_line">record_type_code_of_header_line</a> [<a href="Biocaml_sam.html">Biocaml_sam</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_about.html#VALrelease_date">release_date</a> [<a href="Biocaml_about.html">Biocaml_about</a>]</td>
<td><div class="info">
<code class="code">(year,month,day)</code> of the current release.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Test.html#VALrepeat">repeat</a> [<a href="Biocaml_std.Test.html">Biocaml_std.Test</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_histogram.html#VALreset">reset</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Return histogram with same bins but all counts reset to 0.0.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALrev_fwd">rev_fwd</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td><div class="info">
Return "rev" or "fwd".
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Order.html#VALreverse">reverse</a> [<a href="Biocaml_std.Order.html">Biocaml_std.Order</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Order.html#VALreversep">reversep</a> [<a href="Biocaml_std.Order.html">Biocaml_std.Order</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALrms">rms</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Root mean square.
</div>
</td></tr>
<tr><td><a href="Biocaml_chrName.html#VALroman">roman</a> [<a href="Biocaml_chrName.html">Biocaml_chrName</a>]</td>
<td><div class="info">
<code class="code">roman s</code> returns the canonical Roman representation of <code class="code">s</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALrow">row</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">row m i</code> returns the <code class="code">i</code>th row of matrix <code class="code">m</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALrow_to_string">row_to_string</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALrow_to_string">row_to_string</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td><div class="info">
String representation of row in same format as required by specification.
</div>
</td></tr>
<tr><td align="left"><br>S</td></tr>
<tr><td><a href="Biocaml_bar.html#VALscale">scale</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Return scale data is reported in, e.g.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALsection">section</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
<code class="code">section t name</code> returns the section named <code class="code">name</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALsectioni">sectioni</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
<code class="code">sectioni t i</code> returns the <code class="code">i</code>'th section.
</div>
</td></tr>
<tr><td><a href="Biocaml_bar.html#VALsections">sections</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Return all sections in <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALset">set</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
<code class="code">set t a x</code> sets attribute <code class="code">a</code> to value <code class="code">x</code> in <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALset_attribute">set_attribute</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
<code class="code">set_attribute x y r</code> sets attribute <code class="code">x</code> to <code class="code">y</code> in <code class="code">r</code>, deleting
      any previous values.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALset_col">set_col</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
Set the file name, line number, or column number.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALset_file">set_file</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_pos.html#VALset_line">set_line</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Test.html#VALsf">sf</a> [<a href="Biocaml_std.Test.html">Biocaml_std.Test</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALshuffle">shuffle</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">shuffle arr</code> takes an array and randomly shuffles it.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.StringMap.html#VALsize">size</a> [<a href="Biocaml_std.StringMap.html">Biocaml_std.StringMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_rSet.html#VALsize">size</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Number of elements in set.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALsize">size</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">size v</code> returns the number of integers in <code class="code">v</code>, i.e.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.Selection.html#VALsize">size</a> [<a href="Biocaml_genomeMap.Selection.html">Biocaml_genomeMap.Selection</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALskip_while">skip_while</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALskip_whilei">skip_whilei</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_seq.html#VALslice">slice</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
<code class="code">slice first last t</code> returns the sub-sequence, or slice, of <code class="code">t</code> starting from index <code class="code">first</code> to <code class="code">last</code> (inclusive).
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALspearman">spearman</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">spearman arr1 arr2</code> computes the Spearman rank correlation coefficient of two float arrays.
</div>
</td></tr>
<tr><td><a href="Biocaml_bed.html#VALsqlite_db_of_enum">sqlite_db_of_enum</a> [<a href="Biocaml_bed.html">Biocaml_bed</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALsquish">squish</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALstdv">stdv</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Standard deviation.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALstrict_after">strict_after</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">strict_after u v</code> is equivalent to <code class="code">strict_before v u</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALstrict_before">strict_before</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">strict_before u v</code> is true if <code class="code">u.lo &lt; v.lo &amp;&amp; u.hi &lt; v.hi</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALstrict_subset">strict_subset</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">strict_subset u v</code> is true if <code class="code">u</code> is a strict subset of <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALstrict_superset">strict_superset</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">strict_superset u v</code> is true if <code class="code">u</code> is a strict superset of <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.String.html#VALstrip_final_cr">strip_final_cr</a> [<a href="Biocaml_std.String.html">Biocaml_std.String</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_rSet.html#VALsubset">subset</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
<code class="code">subset s t</code> returns true if <code class="code">s</code> is a subset of <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALsubset">subset</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">subset u v</code> is true if <code class="code">u</code> is a subset of <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALsuperset">superset</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">superset u v</code> is true if <code class="code">u</code> is a superset of <code class="code">v</code>.
</div>
</td></tr>
<tr><td align="left"><br>T</td></tr>
<tr><td><a href="Biocaml_tags.html#VALtag_is">tag_is</a> [<a href="Biocaml_tags.html">Biocaml_tags</a>]</td>
<td><div class="info">
<code class="code">tag_is x y t</code> returns true if <code class="code">x</code>=<code class="code">y</code> in <code class="code">t</code>, or false if tag
      <code class="code">x</code> assigned some other value or not found in <code class="code">t</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.Test.html#VALtest">test</a> [<a href="Biocaml_rSet.Test.html">Biocaml_rSet.Test</a>]</td>
<td><div class="info">
<code class="code">test ul vl</code> compares performance and correctness of set intersection and union.
</div>
</td></tr>
<tr><td><a href="Biocaml_accu.Counter.html#VALtick">tick</a> [<a href="Biocaml_accu.Counter.html">Biocaml_accu.Counter</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Test.html#VALtime">time</a> [<a href="Biocaml_std.Test.html">Biocaml_std.Test</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Test.html#VALtimesf">timesf</a> [<a href="Biocaml_std.Test.html">Biocaml_std.Test</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.List.html#VALtl_exn">tl_exn</a> [<a href="Biocaml_std.List.html">Biocaml_std.List</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALto_ascii">to_ascii</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td><div class="info">
<code class="code">to_ascii ~offset t</code> encodes <code class="code">t</code> as an ASCII character using the
      given offset (default = 33).
</div>
</td></tr>
<tr><td><a href="Biocaml_wig.html#VALto_channel">to_channel</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td><div class="info">
Like <code class="code">to_file</code> but print to channel.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.html#VALto_channel">to_channel</a> [<a href="Biocaml_track.html">Biocaml_track</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALto_channel">to_channel</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_gff.html#VALto_channel">to_channel</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALto_chr_lists">to_chr_lists</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td><div class="info">
Outer list will be in ascending order by <code class="code">chr</code>, and inner lists will be in ascending order by <code class="code">coord</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_wig.html#VALto_file">to_file</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td><div class="info">
<code class="code">to_file ~fmt t file</code> prints <code class="code">t</code> to <code class="code">file</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.html#VALto_file">to_file</a> [<a href="Biocaml_track.html">Biocaml_track</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALto_file">to_file</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td><div class="info">
Items will be printed in ascending order by <code class="code">(chr,coord)</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALto_file">to_file</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALto_file">to_file</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td><div class="info">
<code class="code">to_file file t</code> prints <code class="code">t</code> to <code class="code">file</code> in format required by specification.
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALto_int">to_int</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
Convert roman numeral to integer.
</div>
</td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALto_int">to_int</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.IntSet.html#VALto_list">to_list</a> [<a href="Biocaml_std.IntSet.html">Biocaml_std.IntSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALto_list">to_list</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_wig.html#VALto_list">to_list</a> [<a href="Biocaml_wig.html">Biocaml_wig</a>]</td>
<td><div class="info">
Extract data as a flat list.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALto_list">to_list</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
Return list of attribute=value pairs.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.html#VALto_list">to_list</a> [<a href="Biocaml_track.html">Biocaml_track</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_sgr.html#VALto_list">to_list</a> [<a href="Biocaml_sgr.html">Biocaml_sgr</a>]</td>
<td><div class="info">
Items will be returned in ascending order by <code class="code">(chr,coord)</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_rSet.html#VALto_list">to_list</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Return set of integers as a list.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALto_list">to_list</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">to_list v</code> returns the set of integers contained in <code class="code">v</code>, in ascending order.
</div>
</td></tr>
<tr><td><a href="Biocaml_histogram.html#VALto_list">to_list</a> [<a href="Biocaml_histogram.html">Biocaml_histogram</a>]</td>
<td><div class="info">
Return a list of all bin/count pairs.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALto_list">to_list</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_bpmap.html#VALto_list">to_list</a> [<a href="Biocaml_bpmap.html">Biocaml_bpmap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_bar.html#VALto_list">to_list</a> [<a href="Biocaml_bar.html">Biocaml_bar</a>]</td>
<td><div class="info">
Return the data as a list of triplets (chr,pos,v) representing the chromosome name, probe position, and value.
</div>
</td></tr>
<tr><td><a href="Biocaml_gff.html#VALto_map">to_map</a> [<a href="Biocaml_gff.html">Biocaml_gff</a>]</td>
<td><div class="info">
Partitions annotations by chromosome.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALto_pair">to_pair</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">to_pair t</code> returns the int * int pair.
</div>
</td></tr>
<tr><td><a href="Biocaml_phredScore.html#VALto_probability">to_probability</a> [<a href="Biocaml_phredScore.html">Biocaml_phredScore</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_rSet.html#VALto_range_list">to_range_list</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Return set of integers as a minimal list of non-overlapping ranges in ascending order by their coordinates.
</div>
</td></tr>
<tr><td><a href="Biocaml_table.html#VALto_sqlite">to_sqlite</a> [<a href="Biocaml_table.html">Biocaml_table</a>]</td>
<td><div class="info">
<code class="code">to_sqlite t</code> writes contents of <code class="code">t</code> into a SQLite database, and
      returns the handle to it.
</div>
</td></tr>
<tr><td><a href="Biocaml_pos.html#VALto_string">to_string</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
String representation of a position.
</div>
</td></tr>
<tr><td><a href="Biocaml_msg.Tree.html#VALto_string">to_string</a> [<a href="Biocaml_msg.Tree.html">Biocaml_msg.Tree</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_track.BrowserLines.html#VALto_string">to_string</a> [<a href="Biocaml_track.BrowserLines.html">Biocaml_track.BrowserLines</a>]</td>
<td><div class="info">
Return string representation of browser lines.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALto_string">to_string</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
Return string representing track line in required format.
</div>
</td></tr>
<tr><td><a href="Biocaml_strandName.html#VALto_string">to_string</a> [<a href="Biocaml_strandName.html">Biocaml_strandName</a>]</td>
<td><div class="info">
String representation of a strand.
</div>
</td></tr>
<tr><td><a href="Biocaml_seq.html#VALto_string">to_string</a> [<a href="Biocaml_seq.html">Biocaml_seq</a>]</td>
<td><div class="info">
Return string representation of sequence.
</div>
</td></tr>
<tr><td><a href="Biocaml_romanNum.html#VALto_string">to_string</a> [<a href="Biocaml_romanNum.html">Biocaml_romanNum</a>]</td>
<td><div class="info">
String representation of a roman numeral.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALto_string">to_string</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
String representation of an range, intended only for human legibility.
</div>
</td></tr>
<tr><td><a href="Biocaml_comments.html#VALto_string">to_string</a> [<a href="Biocaml_comments.html">Biocaml_comments</a>]</td>
<td><div class="info">
Return string representation of comment lines.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Order.html#VALtotalify">totalify</a> [<a href="Biocaml_std.Order.html">Biocaml_std.Order</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALtranspose">transpose</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Transpose given matrix <code class="code">m</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_std.Stream.html#VALtruncate">truncate</a> [<a href="Biocaml_std.Stream.html">Biocaml_std.Stream</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_std.html#VALtry_finally_exn">try_finally_exn</a> [<a href="Biocaml_std.html">Biocaml_std</a>]</td>
<td></td></tr>
<tr><td align="left"><br>U</td></tr>
<tr><td><a href="Biocaml_rSet.html#VALunion">union</a> [<a href="Biocaml_rSet.html">Biocaml_rSet</a>]</td>
<td><div class="info">
Set union.
</div>
</td></tr>
<tr><td><a href="Biocaml_range.html#VALunion">union</a> [<a href="Biocaml_range.html">Biocaml_range</a>]</td>
<td><div class="info">
<code class="code">union u v</code> returns the ranges representing the union of <code class="code">u</code> and <code class="code">v</code>.
</div>
</td></tr>
<tr><td><a href="Biocaml_genomeMap.LMap.html#VALunion">union</a> [<a href="Biocaml_genomeMap.LMap.html">Biocaml_genomeMap.LMap</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_genomeMap.LSet.html#VALunion">union</a> [<a href="Biocaml_genomeMap.LSet.html">Biocaml_genomeMap.LSet</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_pos.html#VALunknown">unknown</a> [<a href="Biocaml_pos.html">Biocaml_pos</a>]</td>
<td><div class="info">
Represents an unknown position.
</div>
</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALunset">unset</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
<code class="code">unset t a</code> deletes attribute <code class="code">a</code> from <code class="code">t</code>.
</div>
</td></tr>
<tr><td align="left"><br>V</td></tr>
<tr><td><a href="Biocaml_track.TrackLine.html#VALvalid_wig">valid_wig</a> [<a href="Biocaml_track.TrackLine.html">Biocaml_track.TrackLine</a>]</td>
<td><div class="info">
Returns true if <code class="code">t</code> is a valid wiggle track line.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALvariance">variance</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Variance.
</div>
</td></tr>
<tr><td><a href="Biocaml_about.html#VALversion">version</a> [<a href="Biocaml_about.html">Biocaml_about</a>]</td>
<td><div class="info">
String representation of version.
</div>
</td></tr>
<tr><td align="left"><br>W</td></tr>
<tr><td><a href="Biocaml_msg.html#VALwarn">warn</a> [<a href="Biocaml_msg.html">Biocaml_msg</a>]</td>
<td></td></tr>
<tr><td><a href="Biocaml_math.html#VALwilcoxon_rank_sum">wilcoxon_rank_sum</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
<code class="code">wilcoxon_rank_sum ~alpha=(float) arr1 arr2</code> performs the Wilcoxon rank sum test on two arrays with an optional argument alpha, set to 0.05 by default.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALwilcoxon_rank_sum_to_p">wilcoxon_rank_sum_to_p</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
Performs the wilcoxon rank sum on two float arrays and returns the p-value.
</div>
</td></tr>
<tr><td><a href="Biocaml_math.html#VALwilcoxon_rank_sum_to_z">wilcoxon_rank_sum_to_z</a> [<a href="Biocaml_math.html">Biocaml_math</a>]</td>
<td><div class="info">
As below, except returns a z value.
</div>
</td></tr>
</table><br>
</body>
</html>